rust_snuba/src/processors/profiles.rs

```diff
@@ -0,0 +1,173 @@
+use crate::types::{BytesInsertBatch, KafkaMessageMetadata, BadMessage};
+use rust_arroyo::backends::kafka::types::KafkaPayload;
+use serde::{Deserialize, Serialize};
+use std::convert::TryFrom;
+use uuid::Uuid;
+
+pub fn process_message(
+    payload: KafkaPayload,
+    _metadata: KafkaMessageMetadata,
+) -> Result<BytesInsertBatch, BadMessage> {
+    if let Some(payload_bytes) = payload.payload {
+        let msg: FromProfileMessage = serde_json::from_slice(&payload_bytes).map_err(|err| {
+            log::error!("Failed to deserialize message: {}", err);
+            BadMessage
+        })?;
+        let mut profile_msg: ProfileMessage = msg.try_into()?;
+
+        profile_msg.offset = _metadata.offset;
+        profile_msg.partition = _metadata.partition;
+
+        let serialized = serde_json::to_vec(&profile_msg).map_err(|err| {
+            log::error!("Failed to serialize message: {}", err);
+            BadMessage
+        })?;
+
+        return Ok(BytesInsertBatch {
+            rows: vec![serialized],
+        });
+    }
+    Err(BadMessage)
```

Comment:
```suggestion
        Ok(BytesInsertBatch {
            rows: vec![serialized],
        })
```
(dependent on the suggestion above)